### PR TITLE
Added route deletion logic before deleting network during cleanup

### PR DESCRIPTION
**Pull request by snehala27**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: cloud/services/compute/network.go [hunk 1]
```diff
@@ -103,6 +104,26 @@ func (s *Service) DeleteNetwork() error {
 		}
 	}
 
+	// Delete routes associated with network
+	filterString := fmt.Sprintf("description=k8s-node-route name=%s-*", s.scope.Name())
```
Comment (jzhoucliqr) [inline cloud/services/compute/network.go:108]: how about other routes, like the default routes, are they deleted elsewhere?

File: cloud/services/compute/network.go [hunk 2]
```diff
@@ -103,6 +104,26 @@ func (s *Service) DeleteNetwork() error {
...
+
+	for _, route := range routeList.Items {
+		if strings.HasSuffix(route.Network, network.Name) {
+			s.scope.Info("deleting route ", "route:", route.Name)
+			op, err := s.routes.Delete(s.scope.Project(), route.Name).Do()
+			if err != nil {
+				return errors.Wrapf(err, "failed to delete routes")
+			}
+			if err := wait.ForComputeOperation(s.scope.Compute, s.scope.Project(), op); err != nil {
+				return errors.Wrapf(err, "failed to delete routes")
```
Comment (jzhoucliqr) [inline cloud/services/compute/network.go:122]: this err msg is same as above. maybe -> failed to wait delete routes?

File: cloud/services/compute/network.go (hunk 1 above)
Comment (snehala27) [reply cloud/services/compute/network.go:108]: They are auto-deleted when network is deleted

File: cloud/services/compute/network.go (hunk 2 above)
Comment (snehala27) [reply cloud/services/compute/network.go:122]: ++ack

**Pull request merged**